src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -129770,5 +129770,112 @@ static void M<T>(delegate*<T, void> ptr1, delegate*<T> ptr2)
+        [Fact, WorkItem(44438, "https://github.com/dotnet/roslyn/issues/44438")]
+        public void BadOverride_NestedNullableTypeParameter()
+        {
+            var comp = CreateNullableCompilation(@"
+class A
+{
+    public virtual void M1<T>(C<System.Nullable<T>> x) where T : struct
+    {
+    }
+}
+
+class B : A
+{
+    public override void M1<T>(C<System.Nullable<T?> x)
+    {
+    }
+}
+
+class C<T> {}
+");
+            comp.VerifyDiagnostics(
+                // (11,32): error CS0305: Using the generic type 'C<T>' requires 1 type arguments
+                //     public override void M1<T>(C<System.Nullable<T?> x)
+                Diagnostic(ErrorCode.ERR_BadArity, "C<System.Nullable<T?> x").WithArguments("C<T>", "type", "1").WithLocation(11, 32),
+                // (11,54): error CS1003: Syntax error, ',' expected
+                //     public override void M1<T>(C<System.Nullable<T?> x)
+                Diagnostic(ErrorCode.ERR_SyntaxError, "x").WithArguments(",", "").WithLocation(11, 54),
+                // (11,54): error CS0246: The type or namespace name 'x' could not be found (are you missing a using directive or an assembly reference?)
+                //     public override void M1<T>(C<System.Nullable<T?> x)
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "x").WithArguments("x").WithLocation(11, 54),
+                // (11,55): error CS1003: Syntax error, '>' expected
+                //     public override void M1<T>(C<System.Nullable<T?> x)
+                Diagnostic(ErrorCode.ERR_SyntaxError, ")").WithArguments(">", ")").WithLocation(11, 55),
+                // (11,55): error CS1001: Identifier expected
+                //     public override void M1<T>(C<System.Nullable<T?> x)
+                Diagnostic(ErrorCode.ERR_IdentifierExpected, ")").WithLocation(11, 55),
+                // (11,55): error CS0453: The type 'T?' must be a non-nullable value type in order to use it as parameter 'T' in the generic type or method 'Nullable<T>'
+                //     public override void M1<T>(C<System.Nullable<T?> x)
+                Diagnostic(ErrorCode.ERR_ValConstraintNotSatisfied, "").WithArguments("System.Nullable<T>", "T", "T?").WithLocation(11, 55),
+                // (11,55): error CS0453: The type 'T' must be a non-nullable value type in order to use it as parameter 'T' in the generic type or method 'Nullable<T>'
+                //     public override void M1<T>(C<System.Nullable<T?> x)
+                Diagnostic(ErrorCode.ERR_ValConstraintNotSatisfied, "").WithArguments("System.Nullable<T>", "T", "T").WithLocation(11, 55)
+                );
+        }
+
+        [Fact, WorkItem(44438, "https://github.com/dotnet/roslyn/issues/44438")]
+        public void Override_NestedNullableTypeParameter()
+        {
+            var comp = CreateNullableCompilation(@"
+class A
+{
+    public virtual void M1<T>(int x)
+    {
+    }
+}
+
+class B : A
+{
+    public override void M1<T>(System.Nullable<T?> x)
```

Comment:
The syntax error was a necessary part of the original bug. This test doesn't have the same syntax error but is still labeled as being for that bug. Is this just another sub-case that was fixed?

Reply:
The bug reproduces whether or not the closing angle bracket is missing.

Reply:
Basically these were added to try and understand which conditions trigger the bug or not-- and for instance the Hide test below never triggered the bug. Since they don't serve a distinct purpose in ensuring we don't regress this bug I could just delete all but one of the tests.